test/e2e/common.go

```diff
@@ -146,9 +146,36 @@ func DumpSpecResourcesAndCleanup(ctx context.Context, specName string, clusterPr
 		// that cluster variable is not set even if the cluster exists, so we are calling DeleteAllClustersAndWait
 		// instead of DeleteClusterAndWait
 		framework.DeleteAllClustersAndWait(ctx, framework.DeleteAllClustersAndWaitInput{
-			Client:    client,
+			Client:    clusterClient,
 			Namespace: namespace,
 		}, intervalsGetter(specName, "wait-delete-cluster")...)
+
+		// This is added because Metal3Datas, Metal3DataTemplates and Metal3DataClaims take bit more time to delete, if new test is
```

Comment:
This is a bit long. Perhaps "Waiting for Metal3Datas, Metal3DataTemplates and Metal3DataClaims, as these may take longer time to delete" is enough.

Reply:
done